tests/network/expose.go

```diff
@@ -958,3 +937,60 @@ func resolveNodeIPAddrByFamily(virtClient kubecli.KubevirtClient, sourcePod *k8s
 	}
 	return resolveNodeIp(virtClient, sourcePod, *hostname, ipFamily)
 }
+
+type Option func(cmdArgs []string) []string
+
+func getVMIExposeArgs(vmi *v1.VirtualMachineInstance, opts ...Option) []string {
+	return getExposeArgs("virtualmachineinstance", vmi.Namespace, vmi.Name, opts...)
+}
+
+func getVMIRSExposeArgs(vmrs *v1.VirtualMachineInstanceReplicaSet, opts ...Option) []string {
+	return getExposeArgs("vmirs", vmrs.Namespace, vmrs.Name, opts...)
+}
+
+func getVMExposeArgs(vm *v1.VirtualMachine, opts ...Option) []string {
+	return getExposeArgs("virtualmachine", vm.Namespace, vm.Name, opts...)
+}
+
+func getExposeArgs(resource, namespace, name string, opts ...Option) []string {
+	vmiExposeArgs := []string{
+		expose.COMMAND_EXPOSE,
+		resource, "--namespace", namespace, name,
+	}
+
+	for _, f := range opts {
+		vmiExposeArgs = f(vmiExposeArgs)
+	}
+
+	return vmiExposeArgs
+}
+
+func withServiceName(serviceName string) func(cmdArgs []string) []string {
+	return func(cmdArgs []string) []string {
+		return append(cmdArgs, "--name", serviceName)
+	}
+}
+
+func withPort(port string) func(cmdArgs []string) []string {
```

Comment:
I like that you use the builder pattern. Would it make sense to define all the expose helpers in a dedicated module? So this file then ends up only containing test cases

Reply:
I think it make sense. I moved it to expose_util and I kept it under `network` package. Do you like the name, and do you think it worth a separate package for expose_util.go and expose.go  inside the network directory?